types/mui-datatables/index.d.ts

```diff
@@ -317,52 +577,210 @@ export type MUIDataTableOptions = Partial<{
+     */
     searchProps: React.HTMLAttributes<HTMLInputElement>;
+    /**
+     * Search text placeholder.
+     * {@link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-search/index.js Example}
+     */
     searchPlaceholder: string;
+    /** Search text for the table. */
     searchText: string;
+    /**
+     * Indicates if rows can be selected.
+     * @default multiple
+     */
     selectableRows: SelectableRows;
+    /**
+     * Show/hide the select all/deselect all checkbox header for selectable rows.
+     * @default true
+     */
     selectableRowsHeader: boolean;
+    /**
+     * Hides the checkboxes that appear when selectableRows is set to "multiple" or "single".
+     * Can provide a more custom UX, especially when paired with selectableRowsOnClick.
+     * @default false
+     */
     selectableRowsHideCheckboxes: boolean;
+    /**
+     * Enable/disable select toggle when row is clicked.
+     * When False, only checkbox will trigger this action.
+     * @default false
+     */
     selectableRowsOnClick: boolean;
+    /**
+     * Controls the visiblity of the Select Toolbar.
+     * Options:
+     * - 'replace': Select toolbar replaces default toolbar.
+     * - 'above': Appears above the defualt toolbar.
+     * - 'none': Select Toolbar never appears
+     *
+     * @default replace
+     */
     selectToolbarPlacement: 'replace' | 'above' | 'none';
+    /**
+     * Enable remote data source
+     * @default fale
+     */
     serverSide: boolean;
+    /**
+     * Is called for each filter chip and allows you to place custom props on a filter chip.
+     * {@link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-filter/index.js Example}
+     */
     setFilterChipProps: (colIndex: number, colName: string, data: ReadonlyArray<any[]>) => MUIDataTableChip;
+    /**
+     * Is called for each row and allows you to return custom props for this row based on its data.
+     * {@link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-styling/index.js Example}
+     */
     setRowProps: (row: any[], dataIndex: number, rowIndex: number) => object;
+    /**
+     * Is called for the table and allows you to return custom props for the table based on its data.
+     * {@link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-styling/index.js Example}
+     */
     setTableProps: () => object;
+    /**
+     * Enable/disable sort on all columns.
+     * @default true
+     */
     sort: boolean;
+    /**
+     * Enable/disable alphanumeric sorting of filter lists.
+     * @default true
+     */
     sortFilterList: boolean;
+    /**
+     * Sets the column to sort by and its sort direction.
+     * To remove/reset sorting, input in an empty object.
+     * The object options are the column name and the direction.
+     * @type `{ name: string, direction: 'asc' | 'desc' }`
```

Comment:
```suggestion
     * @shape `{ name: string, direction: 'asc' | 'desc' }`
```

I guess linter doesn't like `@type` so lets just use `@shape` after all then. VSCode will still show it in IntelliSense.

Reply:
I was just going to say the same thing!